gatsby-browser.js

```diff
@@ -1,10 +1,22 @@
 import * as React from 'react'
+import { Link } from 'gatsby'
 import { PrismicPreviewProvider } from 'gatsby-plugin-prismic-previews'
+import { PrismicProvider } from '@prismicio/react'
+
 import { repositoryConfigs } from './src/utils/prismicPreviews'
+import { linkResolver } from './src/utils/linkResolver'
+
 import './src/stylesheets/main.scss'
 
 export const wrapRootElement = ({ element }) => (
-  <PrismicPreviewProvider repositoryConfigs={repositoryConfigs}>
-    {element}
-  </PrismicPreviewProvider>
+  <PrismicProvider
+    linkResolver={linkResolver}
+    internalLinkComponent={({ href, ...props }) => (
+      <Link to={href} {...props} />
```

Comment:
What will be the difference between using the `Link` from 'gatsby' vs from 'prismicio/react'?

Reply:
And, will `PrismicProvider` be a required component from now on in react-based frameworks?

Reply:
> What will be the difference between using the Link from 'gatsby' vs from 'prismicio/react'?

Gatsby's `<Link>` - This is only for internal links. It differs from a normal `<a>` element by allowing client-side transitions. Using `<Link>` is why page transitions are so quick with Gatsby sites. External links must use `<a>`.

Prismic's `<PrismicLink>` - This is for internal links and external links. It is configurable to use any custom component for internal or external links and automatically renders the correct one based on the given URL. By default, it uses `<a>` for both. When using `<PrismicLink>` with Gatsby, it should be configured to use Gatsby's `<Link>`. `<PrismicLink>` is a layer above Gatsby's `<Link>`.

---

> And, will PrismicProvider be a required component from now on in react-based frameworks?

`<PrismicProvider>` technically isn't required since `<PrismicLink>` and `<PrismicRichText>` can be configured directly with props, but it is much easier to configure them once in `<PrismicProvider>`. Using `<PrismicProvider>` in all Prismic + React projects is recommended.